Add GetTransactionInfo in VTadmin API and page

GuptaManan100 opened the pull request:
<!--
  Thank you for your contribution to the Vitess project.
  How to contribute: https://vitess.io/docs/contributing/
  Please first make sure there is an open Issue to discuss the feature/fix suggested in this PR.
  If this is a new feature, please mark the Issue as "RFC".
 -->

<!-- if this PR is Work in Progress please create it as a Draft Pull Request -->

## Description

<!-- A few sentences describing the overall goals of the pull request's commits. -->
<!-- If this is a bug fix and you think the fix should be backported, please write so. -->
This PR adds a VTAdmin API for getting the transaction information added in https://github.com/vitessio/vitess/pull/17131. This PR should be merged after it is https://github.com/vitessio/vitess/pull/17131.

This PR also adds a page to display the following page - 
<img width="1316" alt="Screenshot 2024-11-08 at 3 20 33 PM" src="https://github.com/user-attachments/assets/6db7d6af-723b-4f73-8221-66f7bea84904">


## Related Issue(s)

<!-- List related issues and pull requests. If this PR fixes an issue, please add it using Fixes #????  -->
- #16245 

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

## Deployment Notes

<!-- Notes regarding deployment of the contained body of work. These should note any db migrations, etc. -->

systay commented on go/test/endtoend/cluster/vtadmin_process.go:
```diff
@@ -0,0 +1,235 @@
+/*
+Copyright 2020 The Vitess Authors.
```
date

GuptaManan100 replied:
Fixed

systay commented on go/test/endtoend/vtadmin/main_test.go:
```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2022 The Vitess Authors.
```
date

GuptaManan100 replied:
Fixed

notfelineit commented on go/vt/vtadmin/api.go:
```diff
@@ -1521,6 +1522,25 @@ func (api *API) GetTopologyPath(ctx context.Context, req *vtadminpb.GetTopologyP
+
+	cluster.AnnotateSpan(c, span)
+
+	if !api.authz.IsAuthorized(ctx, c.ID, rbac.KeyspaceResource, rbac.GetAction) {
```
```suggestion
	if !api.authz.IsAuthorized(ctx, c.ID, rbac.ClusterResource, rbac.GetAction) {
```
Cluster resource might be more accurate to your use case!

notfelineit reviewed (approved):
Left one comment!

GuptaManan100 commented:
Thank you for the reviews! I have fixed up all the comments and also the parsing of time. We were passing it in in nanoseconds, but the dayjs was expecting it to be in seconds. I have fixed this now.
